### PR TITLE
Update ResNet to train on CIFAR-10

### PR DESCRIPTION
**Pull request by jekbradbury**

### CODE REVIEW

File: ResNet/data.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    let filepath = "\(directory)/cifar-10-batches-py"
+    let isdir = Bool(path.isdir(filepath))!
+    if !isdir {
+        print("Downloading CIFAR data...")
+        let command = "wget -nv -O- https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz | tar xzf - -C \(directory)"
+        subprocess.call(command, shell: true)
+    }
+}
+
+func loadBatches(from file: String, in directory: String = ".")
```
Comment (rxwei) [inline ResNet/data.swift]: Same comment as https://github.com/tensorflow/swift-models/pull/52#discussion_r262753076.

File: ResNet/data.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    return (Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.0 }),
+    Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.1 }))
+}
+
+func loadTestBatches() -> (Tensor<Int32>, Tensor<Float>) {
+    return loadBatches(from: "test_batch")
+}
+
+extension Dataset where Element == TensorPair<Tensor<Int32>, Tensor<Float>> {
+    init(fromTuple: (Tensor<Int32>, Tensor<Float>)) {
```
Comment (rxwei) [inline ResNet/data.swift]: Same comment as https://github.com/tensorflow/swift-models/pull/52#discussion_r262752641.

File: ResNet/data.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import Python
+import TensorFlow
+
+func maybeDownload(to directory: String = ".") {
```
Comment (rxwei) [inline ResNet/data.swift]: What does "maybe download" mean? What's the behavior when it fails to download? Could you please document it?

**Comment by jekbradbury** [comment]
Certainly what we're doing with data loading is far from ideal, but I think for now it might be better to demonstrate the simplicity of Python interop than to add a lot of error-checking code for the many things that can go wrong.

File: CIFAR/data.swift [hunk 4]
```diff
@@ -13,12 +15,12 @@ func maybeDownload(to directory: String = ".") {
     }
 }
 
-func loadBatches(from file: String, in directory: String = ".")
-    -> (Tensor<Int32>, Tensor<Float>) {
-    maybeDownload(to: directory)
+// Each CIFAR data file is provided as a Python pickle of NumPy arrays
+func loadCIFARFile(filename: String, in directory: String = ".") -> (Tensor<Int32>, Tensor<Float>) {
```
Comment (rxwei) [inline CIFAR/data.swift]: According to the Swift API Design Guidelines, if it's a normal function (not an initializer or factory method), then the first argument label should form a phrase with the function base name when possible.

```suggestion
func loadCIFARFile(named name: String, in directory: String = ".") -> (Tensor<Int32>, Tensor<Float>) {
```

**Pull request merged**